### PR TITLE
Switch to new provider implementation with Speakeasy.

### PR DESCRIPTION
**Pull request by amruth-opal**
History of changes during this iteration can be found at https://github.com/speakeasy-sdks/terraform-provider-opal 

### `opal_group`
- `message_channel_ids` now required (can provide empty list `[]`)
- `on_call_schedule_ids` now required (can provide empty list `[]`)
- `visibility` now required
- `audit_message_channel` => `message_channel_ids` (List of string ids)
- `on_call_schedule` => `on_call_schedule_ids` (List of string ids)
- `visibility_group` => `visibility_group_ids` (List of string ids)
- `manage_resources` => removed in favor of optional declaration of group <> resource relationship
- `resource` => moved to separate resource `opal_group_resource_list`
- `request_configuration` => `request_configurations`. List of configurations with at minimum a default configuration. Optionally specify extra configurations to apply to targeted groups

### `opal_owner`
- `user` (Block list) => `user_ids` (List of strings)
- `user_ids` required instead of optional

### `opal_resource`
- `admin_owner_id` now optional
- `visibility` now required
- `visibility_group` => `visibility_group_ids` (List of string ids)
- `request_configurations` now required
- `request_configuration` => `request_configurations`. List of configurations with at minimum a default configuration. Optionally specify extra configurations to apply to targeted groups

### `opal_resources_users`
- Grant access to a Resource for a specific User

### `opal_group_tag`
- Associate a Group and a Tag

### `opal_resource_tag`
- Associate a Resource and a Tag

### `opal_tag`
- Create an Opal tag to use with other Opal objects

### `opal_tag_user`
- Associate a User and a Tag

## Checklist
- [x] I performed a self-review of my code
- [x] I manually tested my code change (please list details in description)
- [x] I added unit tests 
- [x] I updated the changelog
- [x] I updated the public facing docs

### CODE REVIEW

**Comment by amruth-opal** [comment]
@andrewsy-opal do you want to disable the acceptance tests? We won't be able to merge this until that's resolved since it blocks merge checks with a failure.

**Comment by andrewsy-opal** [comment]
we're deploying from a branch right? @amruth-opal

File: LICENSE.md
Comment (andrewsy-opal) [inline LICENSE.md:3]: @amruth-opal should we update this copyright?

Comment (amruth-opal) [reply LICENSE.md:3]: Is that how that works? We made it from the speakeasy codegen

**Review by andrewsy-opal — APPROVED** [review]

**Pull request merged**
